From 94a03329ee4a397b95acee5e456f69d170328cd8 Mon Sep 17 00:00:00 2001
From: Nicholas Rishel <nrishel@mozilla.com>
Date: Thu, 8 Oct 2026 12:55:27 -0700
Subject: [PATCH] Use TMap for recursive SelectedFields::subfields

SelectedFields contains a TUnorderedMap whose value type is SelectedFields
itself. std::unordered_map is not required to support incomplete value types,
and libstdc++ 11 (used by Firefox's Linux sysroot) rejects it. std::map works
with an incomplete mapped type in practice.
---
 src/compiler/translator/CollectVariables.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/compiler/translator/CollectVariables.h b/src/compiler/translator/CollectVariables.h
index 056ec718f..4277be3e3 100644
--- a/src/compiler/translator/CollectVariables.h
+++ b/src/compiler/translator/CollectVariables.h
@@ -23,7 +23,7 @@ class NameMap;
 
 struct SelectedFields
 {
-    TUnorderedMap<uint32_t, SelectedFields> subfields;
+    TMap<uint32_t, SelectedFields> subfields;
 };
 
 using SamplersStaticallyUsedWithTexelFetch = TUnorderedMap<const TVariable *, SelectedFields>;
-- 
2.43.0

